### PR TITLE
GP11: po enum functions panic reason

### PR DESCRIPTION
**Pull request by cedric-chaumont-st-dev**
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

### CODE REVIEW

**Comment by cedric-chaumont-st-dev** [comment]
Rebased.

File: lib/libutee/tee_api_objects.c [hunk 1]
```diff
@@ -583,14 +589,29 @@ TEE_Result TEE_GetNextPersistentObject(TEE_ObjectEnumHandle objectEnumerator,
...
+	}
 
-	return res;
+	res = utee_storage_next_enum(objectEnumerator, objectInfo, objectID,
+				     objectIDLen);
+
+	if (res == TEE_SUCCESS)
+		goto out;
+err:
+	if (res == TEE_ERROR_ITEM_NOT_FOUND ||
```
Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_objects.c]: As Jens noted in #355, `If you added res == TEE_SUCCESS here you could get rid of the if just above`

File: lib/libutee/tee_api_objects.c [hunk 2]
```diff
@@ -571,7 +574,10 @@ TEE_Result TEE_StartPersistentObjectEnumerator(TEE_ObjectEnumHandle
 
 	res = utee_storage_start_enum(objectEnumerator, storageID);
 
-	if (res != TEE_SUCCESS && res != TEE_ERROR_ITEM_NOT_FOUND)
+	if (res != TEE_SUCCESS &&
```
Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_objects.c]: I would prefer to have always the same kind of programming, that is here:

	if (res == TEE_SUCCESS ||
	    res == TEE_ERROR_ITEM_NOT_FOUND ||
	    res == TEE_ERROR_CORRUPT_OBJECT ||
	    res == TEE_ERROR_STORAGE_NOT_AVAILABLE)
		return res;
    TEE_Panic(0);

Comment (cedric-chaumont-st-dev) [inline lib/libutee/tee_api_objects.c]: Cannot otherwise error related non-void TEE_Panic function.

**Comment by cedric-chaumont-st-dev** [comment]
Rebased.

File: lib/libutee/tee_api_objects.c (hunk 1 above)
Comment (pascal-brand-st-dev) [inline lib/libutee/tee_api_objects.c]: This one is not taken into account

**Comment by cedric-chaumont-st-dev** [comment]
Updated.

**Comment by cedric-chaumont-st-dev** [comment]
Rebased.

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by cedric-chaumont-st-dev** [comment]
Rebased. Tested (STM, ARM Juno).
28701 subtests of which 0 failed
687 test cases of which 0 failed
0 test case was skipped
TEE test application done!

**Pull request merged**